### PR TITLE
security(deps): Bump django version to 2.2.27

### PR DESCRIPTION
**Pull request by mdtro**
#sync-getsentry

Update Django version from 2.2.24 to 2.2.27 to resolve dependabot identified vulnerabilities.

Django 2.2.25 included some breaking changes for us in terms of URL route matching.

> Changed in Django 2.2.25:
> In older versions, a full-match wasn’t required for a route which ends with $.

Some route changes were necessary:
- The _legacy_ catch-all route no longer behaved as we expected, so it's been adjusted.
  - For example in 2.2.24, a HTTP request to `/<org.slug>/<project.slug>/settings/plugins/<plugin_slug>/` would match on the intended catch-all, be routed to React land, and then redirected to `/settings/<org.slug>/projects/<project.slug>/plugins/<plugin_slug>/`.
  - In >=2.2.25, a full-match is now required. The above HTTP request fails to match even on the original catch-all resulting in a HTTP response of 404.
- Routes in the debug_urls were not behaving as expected and have been adjusted.
  - I also opted to use `re_path` (from `django.urls`) rather than `url` (from `django.conf.urls`). The latter is an alias to `re_path` in 2.2.x, but is [deprecated starting from Django 3.1.x](https://docs.djangoproject.com/en/3.2/ref/urls/#url).

This PR supersedes:
- #31720 
- #31098

### CODE REVIEW

File: src/sentry/web/debug_urls.py [hunk 1]
```diff
@@ -65,67 +65,67 @@
...
-    url(r"^debug/mail/assigned/team/$", DebugSelfAssignedTeamEmailView.as_view()),
-    url(
-        r"^debug/mail/codeowners_auto_sync_failure/$", DebugCodeOwnersAutoSyncFailureView.as_view()
+    re_path("^debug/mail/alert", sentry.web.frontend.debug.mail.alert),
+    re_path("^debug/mail/note", DebugNoteEmailView.as_view()),
+    re_path("^debug/mail/new-release", DebugNewReleaseEmailView.as_view()),
+    re_path("^debug/mail/new-user-feedback", DebugNewUserFeedbackEmailView.as_view()),
+    re_path("^debug/mail/assigned/self", DebugSelfAssignedEmailView.as_view()),
+    re_path("^debug/mail/assigned/team", DebugSelfAssignedTeamEmailView.as_view()),
+    re_path("^debug/mail/assigned", DebugAssignedEmailView.as_view()),
```
Comment (wedamija) [inline src/sentry/web/debug_urls.py:74]: Is this re-ordering necessary to make sure this doesn't capture the `self` and `team` paths?

File: src/sentry/web/urls.py [hunk 2]
```diff
@@ -700,6 +700,5 @@
         name="sentry-project-event-redirect",
     ),
     # Legacy
-    # This triggers a false positive for the urls.W002 Django warning
-    url(r"/$", react_page_view),
+    url(r"", react_page_view),
```
Comment (joshuarli) [inline src/sentry/web/urls.py:703]: TIL

**Review by joshuarli — APPROVED** [review]
Very nice.

**Review by wedamija — APPROVED** [review]
This looks fine. I'm a little paranoid about changing urls, so it might be worth pushing this up onto staging and just clicking around for a minute just to make sure nothing obviously breaks

**Comment by evanpurkhiser** [comment]
Should we update more things to `re_path`?

**Comment by mdtro** [comment]
> This looks fine. I'm a little paranoid about changing urls, so it might be worth pushing this up onto staging and just clicking around for a minute just to make sure nothing obviously breaks

Tested in _staging_ and found no issues. :)

**Review by wedamija — APPROVED** [review]

**Pull request merged**